### DISCUSSION TITLE
Does any way to debug server side code in VSCode?

### DISCUSSION THREAD

**Discussion by avnikolaev** [serverpod/serverpod#2170] [Q&A] [answered]
Hi!

I'm facing with missing possibilities to debug endpoint (or anything else server side code). 
How to do this?
I'm using host with Ubuntu, if it's matter.

best regards!

**Comment by bfemmer** [comment, marked as answer]
You should be able to debug your server code in VS Code. Open up the main.dart file (from your **project_server/bin** folder) in VS Code and then start a debug session as you normally would. Then open another terminal, navigate to your flutter app folder, and start it ...

`flutter run -d chrome`

If you've set breakpoints in your endpoints and then called those endpoints from your UI, the code should break...

![image](https://github.com/serverpod/serverpod/assets/19608410/e8740a79-ad92-4a43-829d-186f396d49cc)

Just be mindful that you'll have to work around timeout exceptions.

The host development environment (Ubuntu, in your case) should not matter.

**Reply by avnikolaev** [reply]
Thanks, it works!
